src/librustc_mir/error_codes.rs

```diff
@@ -1128,6 +1128,62 @@ Remember this solution is unsafe! You will have to ensure that accesses to the
 cell are synchronized.
 "##,
 
+E0493: r##"
+A type with a `Drop` implementation was desconstructed when trying to build a
+static type.
+
+Erroneous code example:
+
+```compile_fail,E0493
+enum DropType {
+    A,
+}
+
+impl Drop for DropType {
+    fn drop(&mut self) {}
+}
+
+struct Foo {
+    field1: DropType,
+}
+
+static FOO: Foo = Foo { ..Foo { field1: DropType::A } }; // error!
+```
+
+The problem here is that if the given type or one of its fields implements the
+`Drop` trait, this `Drop` implementation cannot be called during the static
+type initialization which might cause a memory leak. To prevent this issue,
+either instantiate all the static type's fields by hand or use the `lazy_static`
+crate:
+
+```
+enum DropType {
+    A,
+}
+
+impl Drop for DropType {
+    fn drop(&mut self) {}
+}
+
+struct Foo {
+    field1: DropType,
+}
+
+static FOO: Foo = Foo { field1: DropType::A }; // We initialize all fields
+                                               // by hand.
+```
+
+And by using `lazy_static`:
```

Comment:
I'm not sure we should (only?) suggest `lazy_static` here. It feels like playing favorites while crates like https://docs.rs/once_cell/1.1.0/once_cell/ might be more idiomatic Rust these days.

Reply:
Maybe I should just not talk about those crate at all. I'll remove this part.